.github/workflows/test.yml

```diff
@@ -0,0 +1,42 @@
+name: CI
+on:
+  # Trigger the workflow on push or pull request,
+  # but push action is only for the feature branch
+  pull_request:
+    types: [ opened, synchronize, edited, reopened ]
+  push:
+    branches-ignore:
+      - develop
+      - 'release/**'
+jobs:
+  lint_and_test:
+    name: Static code analyze & Unit test
+    runs-on: ubuntu-latest
+    environment: staging
+    steps:
+      # Setup Java environment in order to build the Android app.
+      - uses: actions/checkout@v2.3.2
+      - uses: actions/setup-java@v1
+        with:
+          java-version: '12.x'
+
+      # Setup the flutter environment.
+      - uses: subosito/flutter-action@v1
+        with:
+          channel: 'stable' # 'dev', 'alpha', default to: 'stable'
+          flutter-version: '2.2.3'
+
+      - name: Get flutter dependencies.
+        run: flutter pub get
+
+      - name: Generate project assets (flutter_gen).
+        run: flutter packages pub run build_runner build
```

Comment:
you won't need this, because you will have to commit those gen files too

Reply:
@sleepylee I refused to commit those files as they're generated resource files https://github.com/luongvo/flutter-survey/blob/develop/.gitignore#L76

Reply:
later in your test, you will have the generated mock files as well 🤔  and sometimes you will have gen classes `.g.dart` that generated from this command too. It is also not very friendly for the newcomer if they have to generate all the time and getting something missed out because 1 person may have a different setup at a time...

Reply:
@sleepylee I might consider adding those files into git. But from my view at this time, I don't see the need of adding them into git so that's why I ignored them 🤷‍♂️  thank you for your kind explanation 🙆‍♂️

Reply:
- /lib/gen is not listed: https://dart.dev/guides/libraries/private-files
- https://beltran.work/blog/should-i-add-generated-dart-files-to-git/ pros of adding them: build speed improvements 
- other than that, if you insist, I'd recommend you add the flag `--delete-conflicting-outputs` to make sure, in the future if you mix them - then you won't break the CI